substrate/frame/contracts/src/lib.rs

```diff
@@ -409,6 +410,10 @@ pub mod pallet {
 		#[pallet::constant]
 		type MaxDebugBufferLen: Get<u32>;
 
+		/// The maximum length of the transient storage in bytes.
```

Comment:
This one is a global limit for transient storage including keys, values and journal entries. I agree, description shall be extended.

Reply:
Ideally, the limit would only account for keys and values, but there's a risk that someone might allocate only small values. The minimal key size is 16 bytes with a 0-byte value. The data overhead for each transient storage entry is 48 bytes, meaning the overhead is larger than the content itself. Therefore, we should ensure a safe margin of at least 3 times the content size (Btrremap also has overhead, so 2 times the content size might be sufficient there). Overall, a margin of 6 times the content size is necessary, which poses a challenge because we have limited free memory at runtime.